token/js/src/extensions/extensionType.ts

```diff
@@ -221,3 +231,57 @@ export function getAccountLenForMint(mint: Mint): number {
     const accountExtensions = extensionTypes.map(getAccountTypeOfMintType);
     return getAccountLen(accountExtensions);
 }
+
+// Get new extension length, with current data as input
+export type GetUpdatedVariableExtensionLen = (current: Buffer | null) => number;
+
+export function getNewAccountLenForExtensionLen(
+    info: AccountInfo<Buffer>,
+    address: PublicKey,
+    e: ExtensionType,
+    extensionLen: number | GetUpdatedVariableExtensionLen,
+    programId = TOKEN_2022_PROGRAM_ID
+): number {
+    const mint = unpackMint(address, info, programId);
+
+    const data = getExtensionData(e, mint.tlvData);
+
+    // Need 2 bytes extension discriminator, 2 bytes length
+    let newExtensionLen = 2 + 2;
```

Comment:
You mentioned its a 4 byte discriminator - but I have it as 2. The test cases work which leads me to believe its correct. Should it be 4?
 
https://github.com/solana-labs/solana-program-library/pull/5667#discussion_r1412247485